src/IotHub/IotHub/IotHub/GetAzureRmIotHubKey.cs

```diff
@@ -39,20 +52,26 @@ public class GetAzureRmIotHubKey : IotHubBaseCmdlet
         [Parameter(
             Position = 1,
             Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Name of the Iot Hub")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
         [Parameter(
-            Position = 2,
             Mandatory = false,
             HelpMessage = "Name of the Key")]
         [ValidateNotNullOrEmpty]
         public string KeyName { get; set; }
 
         public override void ExecuteCmdlet()
         {
+            if (ParameterSetName.Equals(ResourceIdParameterSet))
+            {
+                this.ResourceGroupName = IotHubUtils.GetResourceGroupName(this.ResourceId);
+                this.Name = IotHubUtils.GetIotHubName(this.ResourceId);
```

Comment:
@anusapan is `KeyName` not something that we can pull from the provide resource id?

Reply:
No, we can't retrieve `KeyName` from resourceId

Reply:
@anusapan if the key isn't a tracked resource and can't be represented by a resource id, then we shouldn't have a `-ResourceId` parameter -- however, if you'd like, since the resource id currently represents the id of the hub, you can rename the parameter to be `-HubId` instead, which would be a better indicator of what the parameter is.

Reply:
Fixed